includes/wikia/api/DesignSystemApiController.class.php

```diff
@@ -195,4 +203,60 @@ private function isWikiaOrgCommunity() {
 
 		return wfGetBaseDomainForHost( parse_url( $wgServer, PHP_URL_HOST ) ) === $wgWikiaOrgBaseDomain;
 	}
+
+
+	/**
+	 * External API request to IntentX
+	 */
+	public function getFandomShopDataFromIntentX() {
+		global $wgCityId, $wgMemc, $wgFandomShopMap, $wgFandomShopUrl;
+
+		// get id from parameter or default to city id
+		$id = $this->getVal( static::PARAM_ID, $wgCityId );
+
+		$memcKey = wfMemcKey( self::MEMC_PREFIX_FANDOM_STORE, $id );
+		$slicePos = strpos( $memcKey, ':' );
+		// remove previx from memckey
+		$memcKey = substr( $memcKey, $slicePos + 1 );
```

Comment:
wouldn't `wfSharedMemcKey` solve the issue here? I think is generates a key without wiki id but isolates the environments

Reply:
That would be perfect

Reply:
Agreed.